src/python/pants/help/flag_error_help_printer.py

```diff
@@ -41,7 +41,9 @@ def handle_unknown_flags(self, err: UnknownFlagsError):
                 possibilities.update(oshi.collect_scoped_flags())
 
         for flag in err.flags:
-            print(f"Unknown flag {self.maybe_red(flag)} on {err.arg_scope or 'global'} scope")
+            print(
+                f"Unknown flag {self.maybe_red(flag)} in {(err.arg_scope + ' goal') if err.arg_scope else 'global'} context"
```

Comment:
This is more precise phrasing. We don't actually know the scope of the flag, we just know its CLI context (global position or goal position).